File: array/Solution11.java
class Solution11 {
    public int maxArea(int[] height) {
    	int result = 0;
    	if(height.length < 2){
    		return result;
    	}
    	for(int i = 0; i < height.length; i++){
    		for(int j = i + 1; j < height.length; j++){
    			int cur = volume(height, i, j);
    			if(cur > result){
    				result = cur;
    			}
    		}
    	}
    	return result;
    }

    private int volume(int[] height, int left, int right){
    	int width = right - left;
    	int length = Math.min(height[left], height[right]);
    	return width * length;
    }

    public static void main(String[] args){
    	Solution11 s = new Solution11();
    	int[] test1 = new int[]{1,8,6,2,5,4,8,3,7};
    	System.out.println(s.maxArea(test1));
    }
}

File: C++/Solution11.cpp
#include<iostream>
#include<vector>
#include<stack>

using namespace std;

class Solution11 {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1;
        int area = 0;
        while(right > left){
            if(height[left] >= height[right]){
                area = max(area, (right - left) * height[right]);
                right--;
            }
            else{
                area = max(area, (right - left) * height[left]);
                left++;
            }
        }
        return area;
    }
};

int main(){
    pair<int, int> p ={1, 2};
    cout << p.first << p.second << endl;
    return 0;
}